src/ott/solvers/nn/neuraldual.py

```diff
@@ -299,14 +527,59 @@ def step_fn(state_f, state_g, batch):
         )
 
         # do not update state
-        return loss, dist
+        if to_optimize == "both":
+          return loss_f, loss_g, W2_dist
+        elif to_optimize == "f":
+          return loss_f, W2_dist
+        elif to_optimize == "g":
+          return loss_g, W2_dist
+        else:
+          raise ValueError("Optimization target has been misspecified.")
 
     return step_fn
 
-  def to_dual_potentials(self) -> potentials.DualPotentials:
-    """Return the Kantorovich dual potentials from the trained potentials."""
+  def to_dual_potentials(
+      self, finetune_g: bool = True
+  ) -> potentials.DualPotentials:
+    r"""Return the Kantorovich dual potentials from the trained potentials.
+
+    If `g` returns the gradient of the g potential, \nabla_y g,
+    i.e. `g.returns_potential == False`,
```

Comment:
Would remove this line, doesn't render nicely.

Reply:
Updated to:

![image](https://user-images.githubusercontent.com/707462/217833786-b07f5038-3349-4ea6-90d3-ecf3f8342347.png)